### PR TITLE
initial pass to allow svelte 3 support

### PR DESCRIPTION
**Pull request by kentmclean**

### CODE REVIEW

File: src/single-spa-svelte.js [hunk 1]
```diff
@@ -1,55 +1,57 @@
 const defaultOpts = {
-	// required opts
-	component: null,
-	domElementGetter: null,
-	data: null,
```
Comment (kentmclean) [inline src/single-spa-svelte.js:5]: dropped `data` as a required parameter

File: src/single-spa-svelte.js [hunk 2]
```diff
@@ -1,55 +1,57 @@
...
-			target: opts.domElementGetter(),
-			data: opts.data || {},
-		});
-		resolve();
-	});
+  return new Promise((resolve, reject) => {
+    mountedInstances.instance = new opts.component({
+      ...userOpts,
+      target: opts.domElementGetter(),
+      data: opts.data || {},
```
Comment (kentmclean) [inline src/single-spa-svelte.js:53]: spread `userOpts` in then specify `target` and `data` options

File: src/single-spa-svelte.js [hunk 3]
```diff
@@ -1,55 +1,57 @@
...
 }
 
 function unmount(opts, mountedInstances) {
-	return new Promise((resolve, reject) => {
-		mountedInstances.instance.destroy();
-		resolve();
-	});
+  return new Promise((resolve, reject) => {
+    mountedInstances.instance.$destroy
+      ? mountedInstances.instance.$destroy()
```
Comment (blittle) [inline src/single-spa-svelte.js:62]: So the new api is `$destroy`, but to maintain backwards compatibility also revert to `destroy()`?

File: src/single-spa-svelte.js [hunk 4]
```diff
@@ -1,55 +1,57 @@
...
+      : mountedInstances.instance.destroy();
```
Comment (kentmclean) [inline src/single-spa-svelte.js:63]: svelte 3 changed the destroy method, adding `$` at the beginning. Use it if it's there, drop back to non`$` destroy for svelte 2 compatibility

File: src/single-spa-svelte.js [hunk 5]
```diff
@@ -1,55 +1,57 @@
...
-	return new Promise((resolve, reject) => {
-		mountedInstances.instance = new opts.component({
-			target: opts.domElementGetter(),
-			data: opts.data || {},
-		});
-		resolve();
-	});
+  return new Promise((resolve, reject) => {
+    mountedInstances.instance = new opts.component({
+      ...userOpts,
```
Comment (blittle) [inline src/single-spa-svelte.js]: Will `domElementGetter` and other non svelte options get passed int? Is that okay?

File: README.md [hunk 6]
```diff
@@ -35,9 +35,13 @@ export const unmount = [
...
-- `data`: (optional) Data passed to the root component.
+- `component`: (required) The root component that will be rendered. This component should be compiled by svelte and **not** an iife.
+- `domElementGetter`: (required) A function which will return a dom element. The root component will be mounted in this element.
+
+Svelte-specific options
+
+- `anchor`: (optional) A child of the dom element identified by `domElementGetter` to render the component immediately before
+- `hydrate`: (optional) See the svelte [Creating a component](https://svelte.dev/docs#Creating_a_component) documentation
+- `intro`: (optional) If `true`, will play transitions on initial render, rather than waiting for subsequent state changes
+- `props`: (optional) An object of properties to supply to the component
```
Comment (kentmclean) [inline README.md:46]: https://svelte.dev/docs#Creating_a_component for reference. These options will be passed to the svelte component instance in `mount`

File: src/single-spa-svelte.js (hunk 2 above)
Comment (kentmclean) [reply src/single-spa-svelte.js:53]: wondering if we should be more selective with what is passed to the component from a hygiene perspective. I.e., whitelist those object keys that are specified by https://svelte.dev/docs#Creating_a_component

File: src/single-spa-svelte.js (hunk 5 above)
Comment (kentmclean) [reply src/single-spa-svelte.js]: I was pondering that in the following comment :)

File: src/single-spa-svelte.js (hunk 3 above)
Comment (kentmclean) [reply src/single-spa-svelte.js:62]: yes, that's correct

File: src/single-spa-svelte.js (hunk 2 above)
Comment (blittle) [reply src/single-spa-svelte.js:53]: Maybe a blacklist would be easier. Pick the two or three properties that you know are single-spa specific and don't pass them in. That way if a new version of svelte has a new property, we wouldn't need to update single-spa-svelte.

File: src/single-spa-svelte.js [hunk 7]
```diff
@@ -1,55 +1,64 @@
...
-			data: opts.data || {},
-		});
-		resolve();
-	});
+  let svelteOpts = Object.keys(opts).reduce((object, key) => {
+    if (!Object.keys(defaultOpts).includes(key)) {
+      object[key] = opts[key];
+    }
+    return object;
+  }, {});
```
Comment (kentmclean) [inline src/single-spa-svelte.js:47]: `svelteOpts` becomes the passed in `opts` with the elements from `defaultOpts` removed

**Review by blittle — APPROVED** [review]

**Pull request merged**